utils/envvars/parser.go

```diff
@@ -135,6 +145,37 @@ func parseStringDuration(value string) (time.Duration, error) {
 	return d, nil
 }
 
+// GetSAASHost validates SAAS host env variables.
+// returns it if it's valid, otherwise returns defaultSAASHost.
+func GetSAASHost() string {
+	switch {
+	case govalidator.IsURL(os.Getenv(envSAASAuthHost)):
+		h := os.Getenv(envSAASAuthHost)
+		logrus.Warnf("SAAS host changed to %s.", h)
+
+		return h
+	case govalidator.IsURL(os.Getenv(envSAASChecksHost)):
+		h := os.Getenv(envSAASChecksHost)
+		logrus.Warnf("SAAS host changed to %s.", h)
+
+		return h
+	case govalidator.IsURL(os.Getenv(envSAASTelemetryHost)):
+		h := os.Getenv(envSAASTelemetryHost)
+		logrus.Warnf("SAAS host changed to %s.", h)
+
+		return h
+	case govalidator.IsURL(os.Getenv(envSAASHost)):
+		h := os.Getenv(envSAASHost)
+		logrus.Warnf("SAAS host changed to %s.", h)
+
+		return h
```

Comment:
It doesn't provide backward compatibility, if I will specify `PERCONA_TEST_AUTH_HOST` it will affect all SaaS related subsystems, not only `services/platform/`. 

As I see the ticket description changed, so if we want to keep old variables we should check that used only new variable or only old variables, not both.

Reply:
So in all these cases I should set `PERCONA_TEST_SAAS_HOST` and return it?

Reply:
`Use old environment variables if they are defined.`

If understand that line form ticket correctly, then you should return old variable value to the specific service if it is defined.

For example `services/platform/` should get value from `PERCONA_TEST_AUTH_HOST`, `services/checks` from `PERCONA_TEST_CHECKS_HOST` etc
Or if `PERCONA_TEST_SAAS_HOST` specified then every service should receive value form that env.

Reply:
updated GetSAASHost function accordingly to @AlekSi clarification